script.js

```diff
@@ -1,121 +1,145 @@
+    let bookDetails = document.createElement('div');
+    let readStatus = document.createElement('button')
+    let footer = document.createElement('footer');
     
-    let header = document.createElement('header');
-    header.classList.add('title');
-    bookCard.appendChild(header);
     
-    let h1 = document.createElement('h1');
-    h1.textContent = book.title;
-    header.appendChild(h1);
+    bookCard.classList.add('card');
+    library.appendChild(bookCard);
     
-    let details = document.createElement('div');
-    details.classList.add('details');
-    bookCard.appendChild(details);
+    header.classList.add('title');
+    header.textContent = book.title;
     
-    let parAuthor = document.createElement('p');
-    parAuthor.textContent = `Author: ${book.author}`;
-    let parYear = document.createElement('p');
-    parYear.textContent = `Year: ${book.year}`;
-    let parPages = document.createElement('p');
-    parPages.textContent = `Pages: ${book.pages}`;
-    let btnRead = document.createElement('button');
-    btnRead.classList.add('toggle-read');
-    if (book.read) {
-        btnRead.classList.add('read');
-        btnRead.textContent = 'Read'
+    bookDetails.classList.add('content');
+    for (let [key, detail] of Object.entries(book.details)) {
+      let par = document.createElement('p');
+      par.textContent = '';
+      par.textContent = detail;
+      bookDetails.appendChild(par);
     }
-    else {
-        btnRead.classList.add('not-read');
-        btnRead.textContent = 'Not read';
+  
+    readStatus.classList.add('read-status');
+    if (book.read) {
+      readStatus.classList.add('read');
+      readStatus.textContent = 'Read';
+    } else {
+      readStatus.classList.add('not-read');
+      readStatus.textContent = 'Not read';
     }
-    
-    let footer = document.createElement('footer');
+    readStatus.addEventListener('click', toggleRead);
+    bookDetails.appendChild(readStatus);
+
     footer.classList.add('card-footer')
     footer.innerHTML = '<button class="delete-button"><i class="fas fa-trash"></i></button>';
-    bookCard.appendChild(footer);
-    
-    details.appendChild(parAuthor);
-    details.appendChild(parYear);
-    details.appendChild(parPages);
-    details.appendChild(btnRead);
-    
-    let deleteButton = document.querySelectorAll('.delete-button');
-    deleteButton.forEach(button => button.addEventListener('click', deleteCard));
-    let toggleRead = document.querySelectorAll('.toggle-read');
-    toggleRead.forEach(button => button.addEventListener('click',toggleBookRead));
-    closeModal();
+    bookCard.append(header,bookDetails,footer);
+
+    let removeBook = document.querySelectorAll('.delete-button');
+    removeBook.forEach(book => book.addEventListener('click', removeBookCard));
+  });
+};
+
+function removeBookCard(e) {
+  myLibrary = myLibrary.filter(books => books.title !== e.target.parentNode.parentNode.parentNode.firstChild.innerText);
```

Comment:
Two things:

1. How would you deal with this if `myLibrary` would be immutable?
2. Getting the title with the `e.target` is very fragile and hard to reason about. I suggest you pass the book title when calling `removeBookCard` in the event listener. Let's see what solution you come up with.

Reply:
You can also try and use event bubbling to deal with this from `.bookCard`